packages/wp-story-editor/src/media/mediaPickers/library.js

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { getResourceFromMediaPicker } from '@web-stories-wp/story-editor';
```

Comment:
`getResourceFromMediaPicker` is a WordPress-specific function (getting a resource object from the `wp.media` modal) and as such should live in the `wp-story-editor` package (or eventually a shared package with wp-dashboard`).

Reply:
How about we move it to the existing media package?

Reply:
I don't think the media package should contain such WordPress-specific functions. It's currently pretty platform-agnostic and rather provides the low-level utils like `createResource`.

`getImageResourceFromAttachment` and `getResourceFromMediaPicker` are better suited to live in the `wp-story-editor` package (or perhaps `wp-media` or something) as that's where they're going to be used.

----

Aside: we could consider moving `getResourceFromLocalFile()` and `getResourceFromUrl()` to the media package for easier reusability beyond just the editor.

Reply:
I see! 

> `getImageResourceFromAttachment` and `getResourceFromMediaPicker` are better suited to live in the wp-story-editor package (or perhaps `wp-media` or something) as that's where they're going to be used.

Trying to understand media in the project !

By `getImageResourceFromAttachment`, you mean `getResourceFromAttachment` and all its internal functions. 
https://github.com/google/web-stories-wp/blob/47c003efe226f9d2cbe6dec1be52184ee60dc56b/packages/story-editor/src/app/media/utils/getResourceFromAttachment.js#L157-L177

Which is being used to get resources from the response when using `getMedia` and `uploadMedia` API calls.

https://github.com/google/web-stories-wp/blob/47c003efe226f9d2cbe6dec1be52184ee60dc56b/packages/story-editor/src/app/media/local/useContextValueProvider.js#L85-L95
https://github.com/google/web-stories-wp/blob/47c003efe226f9d2cbe6dec1be52184ee60dc56b/packages/story-editor/src/app/media/utils/useMediaUploadQueue/index.js#L305-L311

So we should move `getResourceFromAttachment` to `wp-story-editor` ( later `wp-media` may be ) to update the REST response data from the move API callbacks? 
https://github.com/rtCamp/web-stories-wp/blob/46ebe7a53d9dff35c01b2a386b2a100576dc3038/packages/wp-story-editor/src/api/media.js#L92-L103
https://github.com/rtCamp/web-stories-wp/blob/46ebe7a53d9dff35c01b2a386b2a100576dc3038/packages/wp-story-editor/src/api/media.js#L74-L75